launch/launch/actions/log.py

```diff
@@ -0,0 +1,144 @@
+# Copyright 2025 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Module for the Log action."""
+
+import logging
+import sys
+from typing import List
+import warnings
+
+import launch.logging
+
+from ..action import Action
+from ..frontend import Entity
+from ..frontend import expose_action
+from ..frontend import Parser  # noqa: F401
+from ..launch_context import LaunchContext
+from ..some_substitutions_type import SomeSubstitutionsType
+from ..substitution import Substitution
+from ..utilities import normalize_to_list_of_substitutions
+
+if sys.version_info >= (3, 11):
+    log_levels = logging.getLevelNamesMapping()
+else:
+    # TODO: Remove after Python 3.11+ is minimum support
+    log_levels = {'NOTSET': logging.NOTSET,
+                  'DEBUG': logging.DEBUG,
+                  'INFO': logging.INFO,
+                  'WARNING': logging.WARNING,
+                  'ERROR': logging.ERROR,
+                  'CRITICAL': logging.CRITICAL}
```

Comment:
Very minor, but I'd prefer doing this directly in `Log.execute()` if `log_levels` is not used anywhere else.

You could also just check if `hasattr(logging, 'getLevelNamesMapping')` instead of checking `sys.version`.

Reply:
I typically prefer using sys.version_info since I find it more human readable at a glance when branches can be removed in the future. However, if you prefer `hasattr` I will use it.

Reply:
No, that's fine! Thanks for moving it.